testing/integration/otel_test.go

```diff
@@ -0,0 +1,306 @@
+}
+
+func TestAPMIngestion(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Default,
+		Stack: &define.Stack{},
+		Local: true,
+		OS: []define.OS{
+			// apm server not supported on darwin
+			{Type: define.Linux},
+		},
+	})
+
+	stackVersion := info.KibanaClient.GetVersion()
+	agentVersion, err := version.New(aVersion.Agent)
+	require.NoError(t, err, "failed to get agent version")
+
+	if stackVersion.LessThan(agentVersion) {
+		// at the transition period when new version of stack is not yet released
+		// and we use override with previous version this test would fail with
+		// events being produced by newer APM than stack version
+		t.Skip("agent version needs to be equal to stack version")
+	}
+
+	// prepare agent
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// prepare input
+	agentWorkDir := fixture.WorkDir()
+	fileName := "content.log"
+	fixture.WriteFileToWorkDir(ctx, "", fileName)
+
+	testId := info.Namespace
+
+	apmConfig := fmt.Sprintf(apmOtelConfig, filepath.Join(agentWorkDir, fileName), testId)
+
+	err = fixture.Configure(ctx, []byte(apmConfig))
+	require.NoError(t, err)
+
+	componentsDir, err := aTesting.FindComponentsDir(agentWorkDir)
+	require.NoError(t, err)
+
+	// start apm default config just configure ES output
+	esHost, err := getESHost()
+	require.NoError(t, err, "failed to get ES host")
+	require.True(t, len(esHost) > 0)
+
+	esUsername := os.Getenv("ELASTICSEARCH_USERNAME")
+	require.True(t, len(esUsername) > 0)
+
+	esPass := os.Getenv("ELASTICSEARCH_PASSWORD")
+	require.True(t, len(esPass) > 0)
+
+	esClient := info.ESClient
+	if esClient == nil {
+		esClient, err = getESClient(esHost, esUsername, esPass)
+		require.NoError(t, err, "failed to create Elasticsearch client")
+	}
+
+	esApiKey, err := createESApiKey(esClient, esHost, esUsername, esPass)
+	require.NoError(t, err, "failed to get api key")
+	require.True(t, len(esApiKey) > 1, "api key is invalid %q", esApiKey)
+
+	apmArgs := []string{
+		"run",
+		"-e",
+		"-E", "output.elasticsearch.hosts=['" + esHost + "']",
+		"-E", "output.elasticsearch.api_key=" + esApiKey,
+		"-E", "apm-server.host=127.0.0.1:8200",
+		"-E", "apm-server.ssl.enabled=false",
+	}
+
+	apmPath := filepath.Join(componentsDir, "apm-server")
+	var apmFixtureWg sync.WaitGroup
+	apmFixtureWg.Add(1)
+	apmContext, apmCancel := context.WithCancel(ctx)
+	defer apmCancel()
+	go func() {
+		aTesting.RunProcess(t,
+			apmContext, 0,
+			true, true,
+			apmPath, apmArgs...)
+		apmFixtureWg.Done()
+	}()
+
+	// start agent
+	var fixtureWg sync.WaitGroup
+	fixtureWg.Add(1)
+	go func() {
+		fixture.RunWithClient(ctx, false)
+		fixtureWg.Done()
+	}()
+
+	go func() {
+		// delayed write
+		<-time.After(10 * time.Second)
```

Comment:
Ten seconds delay is better than thirty seconds (as was the case previously), but is there no way to poll for elastic agent logs and look for a specific string or something to know that it's safe to write the file?  

Agent supports a `logs` subcommand.  We could define a corresponding `ExecLogs` method on the Agent test fixture in https://github.com/elastic/elastic-agent/blob/main/pkg/testing/fixture.go and also a `WaitForLogs` function (similar to [`upgradetest.WaitForWatcher`](https://github.com/elastic/elastic-agent/blob/e3c0695757b3799335214ae20af2e847c0d8cc00/testing/upgradetest/watcher.go#L36)) that periodically calls `ExecLogs` under the hood until some timeout or the expected log entry is found.

Reply:
Nice idea with the `LogWatcher`!